### PR TITLE
Update buildbox to push to ECR

### PR DESCRIPTION
**Pull request by logand22**
This PR updates buildbox images to publish to Amazon ECR instead of Quay.io. 

Buildbox images do not need to be mirrored to Quay.io as they aren't customer facing. They are abstracted behind Make targets.

## Testing
Successful buildbox run: https://drone.platform.teleport.sh/gravitational/teleport/14527/1/7

### CODE REVIEW

File: dronegen/buildbox.go [hunk 1]
```diff
@@ -29,12 +29,8 @@ func buildboxPipelineSteps() []step {
 		waitForDockerStep(),
 	}
 
-	for _, name := range []string{"buildbox", "buildbox-arm"} {
+	for _, name := range []string{"buildbox", "buildbox-arm", "buildbox-centos7"} {
 		for _, fips := range []bool{false, true} {
-			// FIPS is unsupported on ARM/ARM64
-			if name == "buildbox-arm" && fips {
-				continue
-			}
```
Comment (fheinecke) [inline dronegen/buildbox.go:37]: What changed here to allow FIPS on ARM?

Comment (logand22) [reply dronegen/buildbox.go:37]: Nothing currently, however the Dockerfile and Makefile targets exist so I added the ability for drone to keep the image up-to-date. The buildbox isn't used yet. I can revert this change if desired, but it doesn't actually start building images just keeps the buildbox image up to date.

Comment (logand22) [reply dronegen/buildbox.go]: I went ahead and removed the buildbox-arm + fips combination and kept it to the original.

File: .cloudbuild/ci/integration-tests.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ options:
 steps:
   # Run the integration tests. Actual content of this job depends on the changes
   # detected in the PR
-  - name: quay.io/gravitational/teleport-buildbox:teleport10
+  - name: public.ecr.aws/gravitational/teleport-buildbox:teleport10
```
Comment (tcsc) [inline .cloudbuild/ci/integration-tests.yaml:15]: Should we change this to 11 while we're here?

**Review by tcsc — APPROVED** [review]

File: .cloudbuild/ci/integration-tests.yaml (hunk 2 above)
Comment (logand22) [reply .cloudbuild/ci/integration-tests.yaml]: Fixed in https://github.com/gravitational/teleport/pull/15058/commits/0f39a01cb8104859d236bbf442a8365991b5b853

**Review by r0mant — APPROVED** [review]

**Pull request merged**